pkg/pillar/cmd/domainmgr/sriov.go

```diff
@@ -0,0 +1,84 @@
+package domainmgr
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/vishvananda/netlink"
+)
+
+func vfCount(ib types.PhysicalIOAdapter) (uint8, error) {
+	v, exists := ib.Cbattr[types.VfCountFieldName]
+
+	if exists {
+		c, err := strconv.Atoi(v)
+		if err != nil {
+			return 0, fmt.Errorf("Cannot convert %s to integer. Value %s, error: %v", types.VfCountFieldName, v, err)
+		} else {
+			return uint8(c), nil
+		}
+	} else {
+		return 0, nil
+	}
+}
+
+func createVF(device string, vfCount uint8) error {
+	f := fmt.Sprint(types.NicLinuxPath, device, types.NumvfsDevicePath)
+	err := os.WriteFile(f, []byte(fmt.Sprint("%n", vfCount)), 0644)
+	if err != nil {
+		return fmt.Errorf("createVF failed. Cannot write to file %s. Error: %s", f, err)
+	}
+
+	return nil
+}
+
+func createAndGetVF(device string, vfCount uint8) (error, *types.VFList) {
+	var vfs *types.VFList
+	var err error
+	const timeout = 5 * time.Second
+	createVF(device, vfCount)
+	for start := time.Now(); time.Since(start) < timeout; {
+		vfs, err = types.GetVf(device)
+		if vfs != nil {
+			break
+		}
+		time.Sleep(time.Second)
+	}
+	return err, vfs
+}
+
+func setupVfLink(iface string, mac string, index uint8) error {
```

Comment:
Shouldn't thus function be called something lime setupVfHardwareAddr since that is what it is setting?

Reply:
Renamed 👍